### ISSUE TITLE
Documentation Improvements

### ISSUE THREAD

**Issue by spydon** [flame-engine/flame#2895]
# What could be improved
The following is valuable feedback that we got from Google:

**Feedback for Flame documentation**
1. It needs an introduction.
    * It lacks an explanation of what Flame is, what it does, how developers benefit.
    * The context for the entire ecosystem is missing.

2. It needs context.
    * Each section of the documentation lacks any “why” regarding the need for or use of the particular section.
    * Without context, it makes the information less approachable.

3. It needs conceptualization.
    * Each section lacks context for either Flutter or game development.
    * This could include references to Flutter via link or explanation as well as those for common game development concepts.

4. It needs to be more atomic.
    * Some pages should be sections (especially Components and Effects).
    * These sections, when broken up, would be easier to find.

5. It has links that need correction.
    * Some, like those at the bottom of Collision Detection, need to be fixed.

6. It has URLs that include URL-encoded characters like spaces.
    * All of the examples have URLs with spaces in them. That should be fixed.

7. It has many accessibility issues.
    * The color contrast between text and background is poor.
    * Many links lack usable text. (That is, they just link on the word “here”.)
    * Form labels and alt text are missing in places.

# Why should this be improved
Because we want good docs.

# Any risks?
No.

# More information
If you would like to work on a big specific part we can split it out to a separate issue, if you just want to do a small fix related to this issue there is no need to create a new issue.

**Comment by siya100** [comment]
hey!!!@spydon
can you assign me this issue as i am interested in working on it

**Comment by spydon** [comment]
> hey!!!@spydon
> can you assign me this issue as i am interested in working on it
> 

This issue is more of a list than one single issue, which part would you like to work on?

**Comment by siya100** [comment]
I would like to work on introduction part

**Comment by spydon** [comment]
> I would like to work on introduction part

Sounds good, your can just tag this issue in the PR when you open it.
Here are some instructions for how to set up the docs locally: 
https://docs.flame-engine.org/latest/development/documentation.html#building-documentation-locally

**Comment by siya100** [comment]
hey!!!
can you please ellaborate as its my first time contributing in documentation like which tech stacks are required.

**Comment by spydon** [comment]
> hey!!! can you please ellaborate as its my first time contributing in documentation like which tech stacks are required.

It says in the instructions I sent.

**Comment by immadisairaj** [comment]
I have started looking at the 5 and 6.

Seems like the links of examples (6) are automatic from the [dashbook](https://github.com/bluefireteam/dashbook) and so the spaces in the links. What could be done? (either change things in dashbook or update text in example to have `_` instead of ` ` (which I don't think is a good way?) )

The issue in 5 - links are not links but texts in md file. So, they need to be fixed (with some understandable text)

**Comment by spydon** [comment]
Good to hear @immadisairaj!
I think the best way forward for those links is to change it in Dashbook, since that project is owned by us too it should be pretty straight forward.

**Comment by immadisairaj** [comment]
I opened up an issue in Dashbook. I think once that is done, we can do (5) and (6) easily at a time.

**Comment by spydon** [comment]
Sounds great @alt-shreya!
Have a look at these instrutions for how to get the docs up and running locally:
https://docs.flame-engine.org/latest/development/documentation.html#building-documentation-locally

**Comment by immadisairaj** [comment]
Could we probably have a checkbox on what all have been done here. Would be easy to see what all are pending.

**Comment by ufrshubham** [comment]
> Could we probably have a checkbox on what all have been done here? Would be easy to see what all are pending.

I've converted it into a checklist, but I am not sure which ones have been already addressed 😅

**Comment by immadisairaj** [comment]
> I've converted it into a checklist, but I am not sure which ones have been already addressed 😅

5 and 6 are done by me. 😊 I am not sure of rest.. 😅

**Comment by mostafaemara** [comment]
is this still open can i work on this?

**Comment by spydon** [comment]
> is this still open can i work on this?

Everything that is not checked off in the list above is still open.
If you feel that you have enough knowledge within any of the topics, feel free to contribute. :)

**Comment by mostafaemara** [comment]
> > is this still open can i work on this?
> 
> Everything that is not checked off in the list above is still open. If you feel that you have enough knowledge within any of the topics, feel free to contribute. :)

i think i can work on the introduction i love this project and i would like to contribute

**Comment by spydon** [comment]
> i think i can work on the introduction i love this project and i would like to contribute

Glad that you like the project! I just checked, the introduction was already updated here https://github.com/flame-engine/flame/pull/3396/files, apparently we forgot to check the box under this issue.
